kythe/go/serving/xrefs/xrefs.go

```diff
@@ -916,12 +1031,7 @@ readLoop:
 					reply.Filtered.Declarations += int64(filtered)
 					stats.addAnchors(&crs.Declaration, p.Group)
 				}
-			case xrefs.IsRefKind(req.ReferenceKind, idx.Kind):
-				if !pageSet.Contains(idx) {
-					reply.Filtered.References += int64(idx.Count)
-					continue
-				}
-				reply.Total.References += int64(idx.Count)
+			case xrefCategoryRef:
```

Comment:
same for these three. The case-code for xrefCategory{Def,Decl,Ref} appears identical; should they be a single case.

Reply:
They affect different categories in the reply.